### PR TITLE
Add main method to capi_image_builder

### PR DESCRIPTION
**Pull request by DavidFair**
- Wires the various steps together
- Adds directory handling
- Openstack steps to upload the image
- Associated tests

We still need to (in a future PR):
- Cleanup a tmp directory if a user specified one
- Push the rebased repo to Github on completion

To test:
Setup a `clouds.yaml` file with the `openstack` cloud, such that `openstack --os-cloud openstack server list` returns something sane

```bash
cd SCD-Openstack-Utils/capi_image_builder
python3 -m venv venv 
source venv/bin/activate
python3 -m pip install requirements.txt
python3 main.py ~/.ssh/id_rsa.pub
# Grab a coffee
```

Note this requires us to use a pub key system so we can later upload the image (not implemented as of this PR)

### CODE REVIEW

File: capi_image_builder/main.py [hunk 1]
```diff
@@ -1,6 +1,53 @@
-# Clone repo
-# Rebase
-# Remove old files
-# Build
-# Push to OpenStack
-# Push to GitHub with new tag
+# TODO Push to GitHub with new tag
+# TODO cleanup the temporary directory
```
Comment (meoflynn) [inline capi_image_builder/main.py:2]: Are these to do in a future PR?

File: capi_image_builder/main.py [hunk 2]
```diff
@@ -1,6 +1,53 @@
...
+import argparse
+
+from builder.args import Args
+from builder.git_steps import prepare_image_repo
+from builder.image_ops import push_new_image
+from builder.packer import build_image
+
+
+def _parse_args() -> Args:
```
Comment (meoflynn) [inline capi_image_builder/main.py:11]: Could we add a description for what this method does?

File: capi_image_builder/main.py [hunk 3]
```diff
@@ -1,6 +1,53 @@
...
+        "--os_version",
+        default="2004",
+        help="The Ubuntu version to build. Default: 2004",
+    )
+
+    args = parser.parse_args()
+    return Args(**vars(args))
+
+
+def main(args: Args):
```
Comment (meoflynn) [inline capi_image_builder/main.py:44]: Same as above

File: capi_image_builder/test/test_git_ops.py [hunk 4]
```diff
@@ -105,7 +135,7 @@ def test_git_rebase_real(_prepared_repo):
 
     # dfbd4fc1dbb2ee1808b17a8fb4d0a5b03417fb5a
     # is the next merge commit upstream, so this should be
-    # present after the rebase
+    # present after the rebase but not before.
     with pytest.raises(GitCommandError):
         _prepared_repo.repo.git.branch(
             "--contains", "dfbd4fc1dbb2ee1808b17a8fb4d0a5b03417fb5a"
```
Comment (meoflynn) [inline capi_image_builder/test/test_git_ops.py:141]: Which repo are we using as part of this unit test?

**Review by meoflynn — CHANGES_REQUESTED** [review]
Looks good to me, just have some very small changes that might be needed.

File: capi_image_builder/main.py (hunk 1 above)
Comment (DavidFair) [reply capi_image_builder/main.py:2]: Yep

**Review by meoflynn — APPROVED** [review]

**Pull request merged**